jx_sqlite/query_table.py

```diff
@@ -345,8 +345,90 @@ def query(self, query):
         return output
 
     def query_metadata(self, query):
-        Log.error("Not implemented yet")
+        frum, query['from'] = query['from'], self
+        schema = self.sf.tables["."].schema
+        query = QueryOp.wrap(query, schema)
+
+        if query.edges or query.groupby:
+            Log.error("Aggregates(groupby or edge) are not supported")
+        where = query.where
+        tableName = None
+        columnName = None
+        if where.op == "eq" and where.lhs.var == "table":
+            tableName = mo_json.json2value(where.rhs.json)
+        elif where.op =="eq" and where.lhs.var =="name":
+            columnName = mo_json.json2value(where.rhs.json)
+        else:
+            Log.error("Only simple filters are expected like: \"eq\" on table and column name")
+        
+        metadata = []
+        result = self.db.query("""SELECT NAME FROM SQLITE_MASTER WHERE type='table' ORDER BY NAME;""")
+        tables = wrap([{k: d[i] for i, k in enumerate(result.header)} for d in result.data])        
+        for table in tables:
+            if table.name.startswith("__") or (tableName != None and tableName != table.name):
+                continue
+            nested_path = [join_field(split_field(tab.name)[1:]) for tab in jx.reverse(tables) if startswith_field(table.name, tab.name)]
+            command = "PRAGMA table_info(" + quote_table(table.name) + ")"
+            columns = self.db.query(command)
+            for cid, name, dtype, notnull, dfft_value, pk in columns.data:
+                if name.startswith("__"):
+                    continue
+                cname, ctype = untyped_column(name)
+                if columnName != None and columnName != cname:
+                    continue
+                if ctype in STRUCT:
+                    ctype = "nested"
+                else:
+                    ctype={"TEXT": "string", "REAL": "number", "INTEGER": "integer"}.get(dtype)
+
+                c ={"table": table.name,
+                    "name": cname,
+                    "type": ctype,
+                    "nested_path": unwraplist(nested_path)
+                }
+                metadata.append(c)
+                
```

Comment:
it is hard to transform from list to cube. But to converting from table is easier

instead of 

                c ={"table": table.name,
                    "name": cname,
                    "type": ctype,
                    "nested_path": unwraplist(nested_path)
                }
                metadata.append(c)

use 

                metadata.append((table.name, cname, ctype, unwraplist(nested_path)))

then

        if query.format == "cube":
            num_rows = len(metadata)
            temp_data = Data()
            for rownum, d in enumerate(metadata):                
                for k, v in d.items(): 
                    if temp_data[k] == None:
                        temp_data[k] = [None] * num_rows                        
                    temp_data[k][rownum] = v

can be written

        if query.format == "cube":
            header = ["table", "name", "type", "nested_path"]
            temp_data = dict(zip(header, zip(*metadata)))

and list can be changed from:

            return Data(
                meta={"format": "list"},
                data=metadata
            )

to

            header = ["table", "name", "type", "nested_path"]
            return Data(
                meta={"format": "list"},
                data=[dict(zip(header, r)) for r in metadata]
            )